h264/parse.go

```diff
@@ -117,19 +117,31 @@ func readMe(r bitio.Reader, chromaArrayType int, mpm macroblockPredictionMode) (
 		return 0, errors.Wrap(err, "error from readUe")
 	}
 
+	// Need to check that we won't go out of bounds with this index.
+	if i2 >= len(codedBlockPattern[i1]) {
+		return 0, errInvalidCodeNum
+	}
+
 	// Macroblock prediction mode selects third index.
 	switch mpm {
 	case intra4x4, intra8x8:
 		i3 = 0
 	case inter:
 		i3 = 1
 	default:
-		return 0, errors.New("invalid macroblock prediction mode")
+		return 0, errInvalidMPM
 	}
 
 	return codedBlockPattern[i1][i2][i3], nil
 }
 
+// Errors used by readMe.
```

Comment:
Are these errors every likely to be exposed outside of the package, i.e., to callers?

Reply:
I'm not sure at this point if they will be exported.